src/Interfaces.ts

```diff
@@ -1,72 +1,112 @@
-export interface Claim<T extends ClaimAttributes = ClaimAttributes> {
+import * as Joi from 'joi'
+import * as jsigs from 'jsonld-signatures'
+
+export enum ClaimType {
+  Identity = 'Identity',
+  Work = 'Work',
+}
+
+export interface Claim {
   readonly '@context'?: any
   readonly id?: string
-
-  readonly issuer?: string
-  readonly issuanceDate?: Date // should replace dateCreated... or created... once we remove protobufs
-  readonly publicKey?: string
-  readonly signature?: string
-  readonly created?: string
+  readonly issuer: string
+  readonly issuanceDate: string
   readonly type: ClaimType
-  readonly attributes: T // TODO: replace attributes with claim, once we remove protobufs
+  readonly 'https://w3id.org/security#proof'?: any
+  readonly claim: object
 }
 
-interface ClaimContext {
-  readonly '@context': {
-    readonly [key: string]: string
-  }
+export interface ClaimContext {
+  readonly [key: string]: unknown
 }
 
-export function isClaim(object: any): object is Claim {
-  // TODO: use joi or protobuf
-  return !!(object.id && object.publicKey && object.signature && object.type && object.attributes && object.created)
+export interface ClaimTypeContexts {
+  readonly [key: string]: ClaimContext
 }
 
 // WARNING: This MUST account for ALL of the attributes in a Claim, except for id, @context, and signature.
 // Otherwise those attributes without context will be left out of the canonized/signed claim.
 // Refer to https://www.w3.org/2018/jsonld-cg-reports/json-ld/#the-context
-export const ClaimContext: ClaimContext = {
-  '@context': {
-    attributes: 'http://schema.org/CreativeWork',
-    author: 'http://schema.org/author',
-    text: 'http://schema.org/text',
-    created: 'http://purl.org/dc/terms/created',
-    dateCreated: 'http://schema.org/dateCreated',
-    datePublished: 'http://schema.org/datePublished',
-    name: 'http://schema.org/name',
-    tags: 'http://schema.org/keywords',
-    type: 'http://schema.org/additionalType',
-    publicKey: 'http://schema.org/Text',
-  },
+export const DefaultClaimContext: ClaimContext = {
+  cred: 'https://w3id.org/credentials#',
+  schema: 'http://schema.org/',
+
+  issuer: 'cred:issuer',
+  issuanceDate: 'cred:issued',
+  type: 'http://schema.org/additionalType',
+  claim: 'http://schema.org/Thing', // The most generic definition in schema.org,
 }
 
-export interface ClaimAttributes {
-  readonly [key: string]: string
+export const DefaultWorkClaimContext: ClaimContext = {
```

Comment:
I see archiveUrl now 👍 but its missing hash?

Reply:
@kennylavender What hashing method are we using in order to verify the content? Or is that something we also allow the customer to specify?

Reply:
oh wait, this is where we want to use multihash.

Reply:
@krobi64 good question, currently we are using [IPFS's default](https://docs.ipfs.io/reference/api/http/#api-v0-add): sha2-256

But I think we want to accept any multihash: https://github.com/multiformats/multihash

is that right @lautarodragan ?